src/Lykke.AzureQueueIntegration/Publisher/AzureQueuePublisher.cs

```diff
@@ -32,18 +43,60 @@ public AzureQueuePublisher(string applicationName, AzureQueueSettings settings)
             _settings = settings;
             _settings.QueueName = _settings.QueueName.ToLower();
 
+            _cloudQueue = _settings.GetQueueAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+            if (disableTelemetry)
+                DisableTelemetry();
+        }
+
+        /// <summary>
+        /// Creates instance of the <see cref="AzureQueuePublisher{TModel}"/>
+        /// </summary>
+        /// <param name="logFactory">Log factory</param>
+        /// <param name="serializer">Serializer</param>
+        /// <param name="publisherName">Name of the publisher</param>
+        /// <param name="settings">Queue settings</param>
+        /// <param name="bufferLifetime">Time interval, which indicates how often buffered messages will be published to the queue</param>
+        /// <param name="disableTelemetry">Disables Application Insight telemetry</param>
+        public AzureQueuePublisher(
+            [NotNull] ILogFactory logFactory,
+            [NotNull] IAzureQueueSerializer<TModel> serializer,
+            [NotNull] string publisherName,
+            [NotNull] AzureQueueSettings settings,
+            TimeSpan? bufferLifetime = null,
+            bool disableTelemetry = true)
+            
+            : base(bufferLifetime ?? TimeSpan.FromMilliseconds(100), logFactory, publisherName)
+        {
+            _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+            _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+
+            _log = logFactory.CreateLog(this, publisherName);
+
+            _settings = new AzureQueueSettings
+            {
+                QueueName = _settings.QueueName.ToLower(),
+                ConnectionString = _settings.ConnectionString
+            };
+
+            _cloudQueue = _settings.GetQueueAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Believe .ConfigureAwait(false) is redundant. Any case GetResult() blocks the calling thread and execution will continue when it release the block.

Reply:
Removed